### PR TITLE
Allow local storage support directory and partition the log files by table

### PR DESCRIPTION
**Pull request by YuJuncen**
This PR make the template of log files become `t{table_id}/{max_ts_YYYYddmm}-{min_ts}-{UUID}.log`.

Also, this PR added subdirectories support for `local` storage backend.

### CODE REVIEW

**Comment by YuJuncen** [comment]
/hold
This PR would make the restore become impossible because of the `ExteranlStorage::restore` doesn't supports files with subdirectories structure.
Possible solution: 
- Bypass the `restore` call, use the `AsyncRead` from `ExternalStorage` directly. (In fact, we don't need to download this to local disk -- no need for ingesting...)]
- Create all subdirectories once we downloading.

File: components/backup-stream/src/router.rs [hunk 1]
```diff
@@ -544,26 +545,44 @@ impl TempFileKey {
...
+        return dt.format("%Y%m%d").to_string();
+        #[cfg(not(feature = "failpoints"))]
+        return dt.format("%Y%m%d");
+    }
+
+    fn path_to_log_file(&self, min_ts: u64, max_ts: u64) -> String {
         format!(
-            // "/v1/t{:012}/{:012}-{}.log",
-            "v1_t{:012}_{:012}-{}.log",
+            "t{:08}/{}-{:012}-{}.log",
```
Comment (3pointer) [inline components/backup-stream/src/router.rs]: How to deal with `version` after this change? when we want to make some breaking change of data format. 🤔

Comment (YuJuncen) [reply components/backup-stream/src/router.rs]: done, version added back to root dir

File: components/backup-stream/src/router.rs [hunk 2]
```diff
@@ -544,26 +545,44 @@ impl TempFileKey {
...
+        format!(
+            "schema-meta/{}-{:012}-{}.log",
+            Self::format_date_time(max_ts),
+            min_ts,
+            uuid::Uuid::new_v4(),
+        )
     }
 
-    fn file_name(&self, min_ts: TimeStamp) -> String {
+    fn file_name(&self, min_ts: TimeStamp, max_ts: TimeStamp) -> String {
```
Comment (joccau) [inline components/backup-stream/src/router.rs:581]: Shall we need make a pr about relative code in `br` during restore ?

**Review by joccau — APPROVED** [review]
rest LGTM.

File: components/backup-stream/src/router.rs (hunk 2 above)
Comment (YuJuncen) [reply components/backup-stream/src/router.rs:581]: Local storage in BR supports subdirectories. And the path of log files are defined in the meta files(which's location is unchanged), so it's probably no changes needing for BR.

**Review by 3pointer — APPROVED** [review]
LGTM

**Pull request merged**